scripts/rust-tests.sh

```diff
@@ -20,7 +20,7 @@ export FM_TEST_USE_REAL_DAEMONS=1
 
 if [ -z "${FM_TEST_ONLY:-}" ] || [ "${FM_TEST_ONLY:-}" = "bitcoind" ]; then
   >&2 echo "### Testing against bitcoind"
-  env RUST_BACKTRACE=1 cargo test -p fedimint-tests -- --test-threads=$(($(nproc) * 2)) "$@"
+  env RUST_BACKTRACE=1 cargo test -p fedimint-tests -- --test-threads=16 "$@"
```

Comment:
Any specific reason for this?

Reply:
Ah was failing with 64 threads on my machine, didn't mean to commit the change...maybe we can automatically set the ulimit when running `nix develop`?

Reply:
Oh, I reverted back to `2 * nproc` BTW. I didn't know it breaks for you, and in CI it is a tiny bit faster...

Is it because not enough memory to run all these tests at once, or something else (ulimit?)?

Yeah, I think we should make it an env variable `${FM_TEST_THREADS:-$(($(nproc) * 2))}` or something.

Reply:
Just the ulimit, I forgot to set it.  CI machines run with 16 normally, but locally I can run with more.